legit.py

```diff
@@ -0,0 +1,53 @@
+from git import Repo
+from git.exc import GitCommandError, InvalidGitRepositoryError
+from typing import Iterable
+
+# parent_commits: Tuple[str, str] = (), author: Tuple[str, str] = author,
+# committer: Tuple[str, str] = committer
+# TODO Consider author and commiter param
+# debug and rewrite clone method and others
+# author = Actor("An author", "katridi@yandex.ru")
+# committer = Actor("A committer", "katridi@yandex.ru")
+
+
+class Repository:
+
+    def __init__(self, path: str) -> None:
+        self.repo = Repo.init(path)
+
+    @classmethod
+    def clone(cls, url: str, path: str) -> str:
+        try:
+            Repo(path)
+        except InvalidGitRepositoryError:
+            Repo.clone_from(url, path)
+        return cls(path)
+
+    def create_branch(self, branch_name: str) -> None:
+        # Check if branch exist in other case create new branch
+        try:
+            self.repo.git.checkout('-b', branch_name)
+        except GitCommandError:
+            self.repo.git.checkout(branch_name)
+
+    def commit(self, meaningful_msg: str) -> None:
```

Comment:
will we also deal with meaningless messages? If not, I suggest to use the whole word as the variable identifier and remove the meaningless adjective

Reply:
Totally